src/pipeline/shutDown/index.ts

```diff
@@ -0,0 +1,59 @@
+import * as core from '@actions/core'
+import { WebhookPayload } from '@actions/github/lib/interfaces'
+import fail from '../../helpers/fail'
+import comment from '../../helpers/comment'
+import { vercelRemoveProjectDeploy } from '../../tenants/vercel'
+import { execCommand } from '../../helpers/execCommand'
+import { formatImage } from '../../helpers/formatImage'
+
+interface shutDownPrams {
+	surgeToken: string
+	teardown: boolean
+	payloadContext: WebhookPayload
+	gitCommitSha: string
+	mountedUrl: string
+	outputUrl: string
+	buildingLogUrl: string
+	vercelToken: string
+	deploymentUrlVercel: string
+}
+
+const shutDown = async ({
+	teardown,
+	payloadContext,
+	mountedUrl,
+	surgeToken,
+	buildingLogUrl,
+	vercelToken,
+	deploymentUrlVercel,
+	outputUrl,
+	gitCommitSha,
+}: shutDownPrams): Promise<void> => {
+	if (teardown && payloadContext.action === 'closed') {
```

Comment:
Salve meu capitão! Como vai?
Acho que valeria colocar essa condição fora da função de `shutDown`. O `shutdown` sempre executando na `main.ts` pode gerar confusões, o bloco de condição traria um externo a função traria um contexto melhor. Então imagino que teríamos algo como:

```suggestion
// prepare.ts

const shouldShutDown = teardown && payloadContext.action === 'closed'

return {
...
shouldShutDown,
}

// main.ts
const { shouldShutDown }	prepare(...)

if (shouldShutDown) {
 return await shutdown(...)
}
```